ilastik/utility/slot_name_enum.py

```diff
@@ -0,0 +1,50 @@
+import enum
+from typing import List, Tuple
+from itertools import chain
+
+class SlotNameEnum(enum.IntEnum):
+    """A map from slot names to their indices within a multislot
+
+    Do note that the ENUM_NAMEs that you pick matter, as they determine
+    the human readable version that you get from instances of this enum:
+
+    e.g a value defined like so:
+        MY_DATA_SLOT = 123
+    will have its slotName rendered as
+        'My Data Slot'
+    """
+
+    @property
+    def slotName(self) -> str:
+        """A 'Human Readable Slot Name' str based on its ENUM_SLOT_NAME"""
+        return self.name.replace('_', ' ').title()
+
+    @classmethod
+    def asNameList(cls) -> List[str]:
+        return [item.slotName for item in cls]
+
+    @classmethod
+    def getFirst(cls) -> int:
```

Comment:
If `SlotNameEnum` always starts from 0 and has consecutive enum values, then `getFirst`, `getLast` and `getNext` could be replaced with `0`, `len(MyEnum)` and `len(MyEnum) + 1`.

Reply:
As mentioned above, they don't have to necessarily start at 0. Still, I hope we can move away from this enum soon enough, but it's more readable than what we have today.